lib/SILGen/SILGenStmt.cpp

```diff
@@ -598,8 +600,10 @@ void StmtEmitter::visitDeferStmt(DeferStmt *S) {
   }
   SGF.visitFuncDecl(deferDecl);
 
-  // Register a cleanup to invoke the closure on any exit paths.
-  SGF.Cleanups.pushCleanup<DeferCleanup>(S->getDeferLoc(), S->getCallExpr());
+  // Register a cleanup to invoke the closure on any exit paths, unless it was
+  // negated.
+  SGF.Cleanups.pushCleanup<DeferCleanup>(S->getDeferLoc(), S->getCallExpr(),
+                                         S->isNegated());
```

Comment:
The semantics here for isNegated() should be not-deferred, e.g. I'd expect an immediate `SGF.emitIgnoredExpr(call)` here in `visitDeferStmt` rather than pushing a cleanup at all.